protobuf/src/stream.rs

```diff
@@ -1179,11 +1196,23 @@ impl<'a> CodedOutputStream<'a> {
     }
 }
 
+impl<'a> Write for CodedOutputStream<'a> {
+    fn write(&mut self, buf: &[u8]) -> IOResult<usize> {
+        self.write_raw_bytes(buf)?;
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> IOResult<()> {
+        self.flush().map_err(Into::into)
```

Comment:
`CodedOutputStream::flush(self)` would be clearer I think to make sure that there is no recursion here.